# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1711492645 -3600
#      Tue Aug 17 11:42:58 2021 +0200
# Node ID 1844e74a1c5039415255eb995d3da20b381667ea
# Parent  a093095462c17c13ecd0061155e95854295946c8
Branding 8: Sign setup.exe and all .exe and .dll files.
* * *
Branding 8: Also sign uninstall\helper.exe

diff --git a/browser/installer/windows/sign.py b/browser/installer/windows/sign.py
new file mode 100644
--- /dev/null
+++ b/browser/installer/windows/sign.py
@@ -0,0 +1,26 @@
+import subprocess, sys
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Set-AuthenticodeSignature -FilePath instgen\\setup.exe -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert)"],
+    stdout=sys.stdout)
+p.communicate()
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird -Filter *.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird\\uninstall -Filter helper.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\uninstall\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\bin -Filter betterbird.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird -Filter *.dll -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
diff --git a/toolkit/mozapps/installer/windows/nsis/makensis.mk b/toolkit/mozapps/installer/windows/nsis/makensis.mk
--- a/toolkit/mozapps/installer/windows/nsis/makensis.mk
+++ b/toolkit/mozapps/installer/windows/nsis/makensis.mk
@@ -62,8 +62,10 @@ ifdef MOZ_STUB_INSTALLER
 endif
 
 ifdef ZIP_IN
 installer:: $(CONFIG_DIR)/setup.exe $(ZIP_IN)
+	@echo 'Signing.'
+	$(PYTHON3) $(topsrcdir)/browser/installer/windows/sign.py
 	@echo 'Packaging $(WIN32_INSTALLER_OUT).'
 	$(NSINSTALL) -D '$(ABS_DIST)/$(PKG_INST_PATH)'
 	$(PYTHON3) $(MOZILLA_DIR)/mach repackage installer \
 	  -o '$(ABS_DIST)/$(PKG_INST_PATH)$(PKG_INST_BASENAME).exe' \
